core/src/main/java/hudson/model/Queue.java

```diff
@@ -2443,29 +2485,45 @@ protected NotWaitingItem(NotWaitingItem ni) {
      * {@link Item} in the {@link Queue#blockedProjects} stage.
      */
     public final class BlockedItem extends NotWaitingItem {
+        private transient CauseOfBlockage causeOfBlockage = null;
+
+        /**
+         * @deprecated Use {@link #BlockedItem(WaitingItem, CauseOfBlockage)} instead
+         */
+        @Deprecated
         public BlockedItem(WaitingItem wi) {
-            super(wi);
+            this(wi, null);
         }
 
+        /**
+         * @deprecated Use {@link #BlockedItem(NotWaitingItem, CauseOfBlockage)} instead
+         */
+        @Deprecated
         public BlockedItem(NotWaitingItem ni) {
+            this(ni, null);
+        }
+
+        public BlockedItem(WaitingItem wi, CauseOfBlockage causeOfBlockage) {
```

Comment:
Why would this be `public`? It is only called from within the same source file AFAIK.